doc/plantuml/provisioning_operation_highlevel_design.pu

```diff
@@ -0,0 +1,73 @@
+@startuml
+
+title High-Level Diagram of an Operation in Provisioning Library
+
+skinparam classFontSize 8
+skinparam classFontName Helvetica
+
+box "User Application" #Wheat
+actor "Application Task" as app
+participant "User Callback\n(for Provisioning operation)" as callback
+end box
+
+box "Provisioning Library" #LightBlue
+participant "Provisioning" as provisioning
+end box
+
+box "MQTT Library" #LightGreen
+participant "MQTT" as mqtt
+end box
+
+box "AWS" #LightPink
+participant "AWS IoT Core" as aws
+end box
+
+activate app
+app -> mqtt: Create MQTT connection
+mqtt -> app: Return MQTT connection handle or error.
+note right of app: Recommended to <i>share</i> MQTT connection\nacross Provisioning API calls.
+
+app -> provisioning: Call Provisioning operation API.\n Provide user-callback for operation.
+
+provisioning -> provisioning: Validate input arguments
+provisioning --> app: Return BAD_PARAMETER for any invalid input argument
+
+provisioning -> mqtt: Subscribe to 'accepted' and 'rejected' response topics\nof operation
+
+mqtt -> aws: Transmit MQTT SUBSCRIBE packets to Server
+mqtt -> provisioning: Return SUCCESS or error
+provisioning --> app: Return MQTT_ERROR if error in subscribing
+
+provisioning -> provisioning: Serialize PUBLISH payload
+provisioning -> provisioning: Return error if serialization failed.
```

Comment:
does this error need to be passed to the Application?

Reply:
Updated the comment to be more meaningful about the error values that are passed for serialization failure back to the application.
Also, fixed the direction of the arrow to point from `provisioning` -> `app`